lib/ansible/modules/network/eos/eos_user.py

```diff
@@ -47,6 +47,12 @@
         device.  This argument accepts a stringv value and is mutually
         exclusive with the C(users) argument.
         Please note that this option is not same as C(provider username).
+  password:
+    description:
+      - The password to be configured on the remote Arista EOS device. The
+        password needs to be provided in clear and it will be encrypted
+        on the device.
+        Please note that this option is not same as C(provider username).
```

Comment:
ah this would be `C(provider password)` in this case.
`username` was specifically for `username` option. 
Please change it to "Please note that this option is not same as C(provider password)."